pkg/util/throughput-cal/throughput-cal.go

```diff
@@ -0,0 +1,58 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package throughput_cal
```

Comment:
This is not an idiomatic way to name a package. Take a look at this: 
https://golang.org/doc/effective_go#package-names

You are renaming the package import everywhere, why not call it `throughput`?

Also, one file packages can cause a lot of stutter in the code if not careful.
Examples: 
`throughput.InitThroughputWatcher` vs. `throughput.InitWatcher`
`throughput.NewThroughputCal` vs. `throughput.NewCalc`